pkg/services/sqlstore/migrations/ualert/ualert.go

```diff
@@ -451,3 +432,133 @@ type rmMigrationWithoutLogging = rmMigration
 func (m *rmMigrationWithoutLogging) SkipMigrationLog() bool {
 	return true
 }
+
+type upgradeNgAlerting struct {
+	migrator.MigrationBase
+}
+
+var _ migrator.CodeMigration = &upgradeNgAlerting{}
+
+func (u *upgradeNgAlerting) Exec(sess *xorm.Session, migrator *migrator.Migrator) error {
+	firstOrgId, err := u.updateAlertConfigurations(sess, migrator)
+	if err != nil {
+		return err
+	}
+	u.updateAlertmanagerFiles(firstOrgId, migrator)
+	return nil
+}
+
+func (u *upgradeNgAlerting) updateAlertConfigurations(sess *xorm.Session, migrator *migrator.Migrator) (int64, error) {
+	// if there are records with org_id == 0 then the feature flag was enabled before 8.2 that introduced org separation.
+	// if feature is enabled in 8.2 the migration "AddDashAlertMigration", which is effectively different from what was run in 8.1.x and earlier versions,
+	// will handle organizations correctly, and, therefore, nothing needs to be fixed
+	count, err := sess.Table(&AlertConfiguration{}).Where("org_id = 0").Count()
+	if err != nil {
+		return 0, fmt.Errorf("failed to query table alert_configuration: %w", err)
+	}
+	if count == 0 {
+		return 0, nil // NOTHING TO DO
+	}
+
+	orgs := make([]int64, 0)
+	// get all org IDs sorted in ascending order
+	if err = sess.Table("org").OrderBy("id").Cols("id").Find(&orgs); err != nil {
+		return 0, fmt.Errorf("failed to query table org: %w", err)
+	}
+	if len(orgs) == 0 { // should not really happen
+		migrator.Logger.Info("No organizations are found. Nothing to migrate")
+		return 0, nil
+	}
+
+	firstOrg := orgs[0]
+
+	// assigning all configurations to the first org because 0 does not usually point to any
+	migrator.Logger.Info("Assigning all existing records from alert_configuration to the first organization", "org", firstOrg)
+	_, err = sess.Cols("org_id").Update(&AlertConfiguration{OrgID: firstOrg})
```

Comment:
Do we need a `.Where("org_id = 0")` condition here too?

Reply:
yes, makes sense!